### ISSUE TITLE
No field to link node to!

### ISSUE THREAD

**Issue by kristian76** [amark/gun#211]
When I run this code snippet

```
var gun = Gun('http://localhost:9000/gun');

var project = gun.get('project');

var user = gun.get('user/kristian76').put({
    username: 'kristian76'
});

var foo = gun.get('task/foo').put({
    title: 'Foo'
});
    foo.path('author').put(user);

var bar = gun.get('task/bar').put({
    title: 'Bar'
});
    bar.path('author').put(user);

var baz = gun.get('task/baz').put({
    title: 'Baz'
});
    baz.path('author').put(user);

var foobar = gun.get('task/foobar').put({
    title: 'Foo Bar'
});
    foobar.path('author').put(user);

var tasks = project.path('tasks');

    tasks.set(foo);
    tasks.set(bar);
    tasks.set(baz);
    tasks.set(foobar);

tasks.map().val(function(task) {
    console.log('Task obj', task);
});

gun.get('task/baz').path('author').val(function(author) {
    console.log(author);    
});
```

the console shows 'Task obj Object {_: Object, title: "Foo", author: Object}' etc as expected. If I refresh the page I see the output below

> No field to link node to!
> gun.js:1129No field to link node to!
> gun.js:1129No field to link node to!
> gun.js:1129No field to link node to!
> main.js:61Task obj Object {_: Object, title: "Foo", author: Object}
> main.js:61Task obj Object {_: Object, title: "Bar", author: Object}
> main.js:61Task obj Object {_: Object, title: "Baz", author: Object}
> main.js:61Task obj Object {_: Object, title: "Foo Bar", author: Object}
> main.js:65Object {_: Object, username: "kristian76", #: "user/kristian76"}
> (4) gun.js:1129 No field to link node to!
> (4 )gun.js:1129 No field to link node to!
> (4) gun.js:1129 No field to link node to!
> (4) gun.js:1129 No field to link node to!
> (4) gun.js:1129 No field to link node to!
> (4) gun.js:1129 No field to link node to!
> (12) gun.js:1129 No field to link node to!

**Comment by anrid** [comment]
I have the same problem when running the following;

```
'use strict'

const Assert = require('assert')
const P = require('bluebird')
const Gun = require('gun')

const gun = Gun()

// Create a family.
const dad = gun.get(`user/dad`).put({ name: 'Dad', age: 50, email: 'dad@example.com' })
const mom = gun.get(`user/mom`).put({ name: 'Mom', age: 45, email: 'mom@example.com' })
const boy = gun.get(`user/boy`).put({ name: 'Boy', age: 18, email: 'boy@example.com' })
const girl = gun.get(`user/girl`).put({ name: 'Girl', age: 16, email: 'girl@example.com' })

// Fetch dad.
gun.get('user/dad').val(node => console.log('Got user:', node.name))

// Create a set of all users.
const users = gun.get('users')

// Show email addresses of all users and subscribe to future changes.
users.map().path('email').val('User’s')

users.set(dad)
users.set(mom)
users.set(boy)
users.set(girl)

// Create a company.
const company = gun.get('startup/hype').put({
  name: 'hype',
  profitable: false,
  address: {
    street: '123 Hipster Lane',
    city: 'San Francisco',
    state: 'CA',
    country: 'USA'
  }
})

company.path('address.city').val((value, field) => console.log(`What’s the ${field}? ${value}`))

// Update company info.
company.put({
  funded: true,
  profitable: true,
  address: {
    street: '999 Expensive Boulevard'
  }
}, () => company.path('address.street').val((value, field) => console.log(`What’s the ${field}? ${value}`)))

// Add various relationships.
// Dad and Mom are both employees and Mom’s the CEO.
const employees = company.path('employees')
employees.set(dad)
employees.set(mom)

company.path('ceo').put(mom)
company.path('ceo').path('email').val('Startup CEO')

dad.path('employeer').put(company)
mom.path('employeer').put(company)

// Dad and Mom are married (for now).
dad.path('spouse').put(mom)
mom.path('spouse').put(dad)

// Dad and Mom are the parents of Boy and Girl.
dad.path('children').set(boy)
dad.path('children').set(girl)

mom.path('children').set(boy)
mom.path('children').set(girl)

boy.path('dad').put(dad)
boy.path('mom').put(mom)

girl.path('dad').put(dad)
girl.path('mom').put(mom)

// Fetch Dad’s spouse’s employer’s employees
const p1 = new P(resolve => (
  gun.get('user/dad').path('spouse.employeer.employees').val(resolve)
))

// Fetch Girl’s dad’s employer’s employees
const p2 = new P(resolve => (
  gun.get('user/girl').path('dad.employeer.employees').val(resolve)
))

P.all([p1, p2]).spread((r1, r2) => {
  Assert.deepStrictEqual(r1, r2)
  console.log('All is well in the world.')
})
```

First run without a `data.json` created runs fine;

```
$ rm data.json; node gundb/
Hello wonderful person! :) I'm mark@gunDB.io, message me for help or with hatemail. I want to hear from you! <3
WARNING! This `file.js` module for gun is intended only for local development testing!
Got user: Dad
User’s email: girl@example.com
User’s email: boy@example.com
User’s email: mom@example.com
User’s email: dad@example.com
Startup CEO email: mom@example.com
What’s the city? San Francisco
All is well in the world.
What’s the street? 999 Expensive Boulevard
```

Subsequent calls go nuts;

```
$ node gundb/ 
Hello wonderful person! :) I'm mark@gunDB.io, message me for help or with hatemail. I want to hear from you! <3
WARNING! This `file.js` module for gun is intended only for local development testing!
Got user: Dad
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
User’s email: girl@example.com
User’s email: boy@example.com
User’s email: mom@example.com
User’s email: dad@example.com
No field to link node to!
No field to link node to!
What’s the city? San Francisco
No field to link node to!
No field to link node to!
No field to link node to!
Startup CEO email: mom@example.com
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
No field to link node to!
All is well in the world.
What’s the street? 999 Expensive Boulevard
```

**Comment by amark** [comment]
Hey guys, sorry I guess I didn't notice this till now.

Let me re-read the issue, but while you are doing that. Could you try this on the `0.5` branch? 0.5 is not done yet (about 60% coverage), but weird quirks like this should be fixed. Plus you will get the performance improvements.

Doing a more detailed read through your comments now...

**Comment by amark** [comment]
@kristian76 doing a quick test with it on `0.5` it seemed to work. So `0.5` has the fix to this bug in `0.3`.

@anrid just as a note, `0.5` probably won't support the `.val("prefix message")`.

@anrid when I tried running your example with `0.5` it did not work yet. :( However, good news on `0.3`....

@anrid on current/master `0.3` I do get a bunch of crazy messages, but at least the output of the two lists are correct and match (notice that the "the list?" and "the same list?" both are the same):

```
"Got user:"
"Dad"
"User’s"
"email:"
"dad@example.com"
"User’s"
"email:"
"mom@example.com"
"User’s"
"email:"
"boy@example.com"
"User’s"
"email:"
"girl@example.com"
"What’s the city? San Francisco"
"No field to link node to!"
"Startup CEO"
"email:"
"mom@example.com"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"No field to link node to!"
"the list?"
"employees:"
[object Object] {
  _: [object Object] {
    #: "riXJVg7FuEs1k6OFZouuzkjB",
    >: [object Object] { ... }
  },
  user/dad: [object Object] {
    #: "user/dad"
  },
  user/mom: [object Object] {
    #: "user/mom"
  }
}
"the same list?"
"employees:"
[object Object] {
  _: [object Object] {
    #: "riXJVg7FuEs1k6OFZouuzkjB",
    >: [object Object] { ... }
  },
  user/dad: [object Object] {
    #: "user/dad"
  },
  user/mom: [object Object] {
    #: "user/mom"
  }
}
"Warning! You have no peers to connect to!"
"What’s the street? 999 Expensive Boulevard"
```

@anrid this is what I used to test: http://jsbin.com/puweyuyisu/edit?html,js,console .

@kristian76 and @anrid do NOTE that on a 2nd run, you are re-saving the exact same data (which is fine, because you are re-using the same context/reference so it should work fine) which is not something that happens in real apps.

There definitely seems to be something weird/problematic going on when the exact same data gets re-saved. This will get fixed in `0.5` but probably not in `0.3` :( sorry. But it looks like none of the actual data is getting corrupted.

Please hit me up if you have any other problem/complaints. I'm going to leave this issue open until `0.5` has 100% coverage and is merged into master/stable.

**Comment by amark** [comment]
Just tried this out again (had to change the `.val(string)` usage which 0.5 no longer has) and it seemed to work. So closing! :)
